legal-api/src/legal_api/services/warnings/business/business_checks/involuntary_dissolution.py

```diff
@@ -52,11 +56,35 @@ def check_business(business: Business) -> list:
             result.append(transition_warning)
         elif dis_details.ar_overdue:
             result.append(ar_overdue_warning)
+
+        data = _get_modified_warning_data(batch_processing)
+
         result.append({
             'code': BusinessWarningCodes.DISSOLUTION_IN_PROGRESS,
-            'data': batch_processing.meta_data,
+            'data': data,
             'message': 'Business is in the process of involuntary dissolution.',
             'warningType': WarningType.INVOLUNTARY_DISSOLUTION
         })
 
     return result
+
+
+def _get_modified_warning_data(batch_processing: BatchProcessing) -> dict:
+    """Return involuntary disssolution warning data based on rules."""
+    meta_data = batch_processing.meta_data if batch_processing.meta_data else {}
+
+    trigger_date = batch_processing.trigger_date
+    current_date = datetime.utcnow()
+    modified_target_date = None
+    if batch_processing.step == BatchProcessing.BatchProcessingStep.WARNING_LEVEL_1:
+        modified_target_date = max(current_date, trigger_date) + datedelta(days=current_app.config.get('STAGE_2_DELAY'))
+    elif batch_processing.step == BatchProcessing.BatchProcessingStep.WARNING_LEVEL_2:
+        modified_target_date = max(current_date, trigger_date)
+
+    if modified_target_date:
+        meta_data = {
+            **meta_data,
+            'targetDissolutionDate': modified_target_date.date().isoformat()
```

Comment:
Been thinking about this more...think we should remove the existing logic for populating `targetDissolutionDate` into `batch_processing.meta_data`.  Seems like it would cause more confusion with the logic we are implementing here for the `targetDissolutionDate` value.

Can you update the dissolution job to remove this logic?

Reply:
thanks for the updates.  i totally forgot about COA had updates to `targetDissolutionDate` too.

I think we can remove this line too https://github.com/bcgov/lear/blob/e8cbdc96ad2fa83958505bc3b74652e0ccdc4009/jobs/involuntary-dissolutions/involuntary_dissolutions.py#L203